common/src/types/v0/message_bus/replica.rs

```diff
@@ -47,6 +49,89 @@ impl Replica {
     pub fn online(&self) -> bool {
         self.status.online()
     }
+    /// check if it was created by the control plane
+    pub fn ours(&self) -> bool {
+        let base_name = ReplicaName::new(&self.uuid, None);
+        self.name.0.starts_with(base_name.as_str())
+    }
+}
+
+bus_impl_string_uuid!(ReplicaId, "UUID of a mayastor pool replica");
+
+/// Name of a Replica
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+pub struct ReplicaName(String);
+
+impl ReplicaName {
+    fn base_name() -> String {
+        "mayastor-cp".to_string()
+    }
+    fn trim(src: String, len: u8) -> String {
+        if src.len() >= len as usize {
+            src[0 .. len as usize].to_string()
+        } else {
+            src
+        }
+    }
+    /// Derive Self from option or the replica_uuid, todo: needed until we fix CAS-1107
+    pub fn from_opt_uuid(opt: Option<&Self>, replica_uuid: &ReplicaId) -> Self {
+        opt.unwrap_or(&Self::from_uuid(replica_uuid)).clone()
+    }
+    /// Derive Self from a replica_uuid, todo: needed until we fix CAS-1107
+    pub fn from_uuid(replica_uuid: &ReplicaId) -> Self {
+        // CAS-1107 -> replica uuid are not checked to be unique, so until that is fixed
+        // use the name as uuid (since the name is guaranteed to be unique)
+        ReplicaName(replica_uuid.to_string())
+    }
+    /// Create new `Self` derived from the replica and volume id's
+    pub fn new(replica_uuid: &ReplicaId, _volume_uuid: Option<&VolumeId>) -> Self {
+        // CAS-1107 -> replica uuid are not checked to be unique, so until that is fixed
+        // use the name as uuid (since the name is guaranteed to be unique)
+        ReplicaName(replica_uuid.to_string())
+    }
+    /// Create new `Self` derived from the replica and volume id's
+    #[allow(dead_code)]
+    fn derive_name(replica_uuid: &ReplicaId, volume_uuid: Option<&VolumeId>) -> Self {
+        let base = Self::base_name();
+        assert!(base.len() < 15);
+
+        let replica_uuid = Self::trim(replica_uuid.to_string(), 22);
+        let name = if let Some(volume_uuid) = volume_uuid {
+            let volume_uuid = Self::trim(volume_uuid.to_string(), 22);
```

Comment:
Where do these character limits come from? I tried looking at the spec but couldn't figure it out.